integrated_channels/canvas/client.py

```diff
@@ -273,18 +331,26 @@ def _extract_integration_id(self, data):
 
         return integration_id
 
-    def _get_course_id_from_integration_id(self, integration_id):
+    def _find_course_by_course_id(self, edx_course_id):
         """
-        To obtain course ID we have to request all courses associated with the integrated
-        account and match the one with our integration ID.
-
-        Args:
-            integration_id (string): The ID retrieved from the transmission payload.
+        Search course by edx_course_id (used as integration_id in canvas) under given account.
+          Issue with this is that if this course exists in canvas with this edx_course_id
+          under a different subaccount, we will not detect it.
+          Hence we will try to find a course under other subaccounts in case we don't find it
+          under the current subaccount
+          Will even return courses that are in the 'deleted' state in Canvas, so we can correctly
+          skip these courses in logic for create_or_update when needed.
+
+          Note: we do not need to follow pagination here since it would be extremely unlikely
+          that searching by a specific edx_course_id results in many records, we generally only
+          expect 1 record to come back anyway.
+
+          The `&state[]=all` is added so we can also fetch priorly 'delete'd courses
         """
-        url = "{}/api/v1/accounts/{}/courses/?search_term={}".format(
+        url = "{}/api/v1/accounts/{}/courses/?search_term={}&state[]=all".format(
```

Comment:
just later added this since without this param, 'deleted' courses will by default be excluded, and I have a clause elsewhere to specifically detect and log deleted courses